### ISSUE TITLE
Documentation layout rework

### ISSUE THREAD

**Issue by sgomes** [google/material-design-lite#1824]
Our pages are too long-running, and the content is poorly organised. We should prepare some mockups and revamp the general layout. Some ideas to consider:

- One page per component, rather than one page per category of components.
- Sub-navigation inside each page, to separate examples from class reference and JSDoc reference.
- Better usage of anchors to save history between components.
- Better layout for displaying examples, to avoid exceeding screen width in mobile.

**Comment by Garbee** [comment]
Also let's make it so headers all get a link symbol when hovered to get the link to that section. This will save people from needing to scroll up to the ToC to get the link to a section to share.

**Comment by addyosmani** [comment]
> Also let's make it so headers all get a link symbol when hovered to get the link to that section. This will save people from needing to scroll up to the ToC to get the link to a section to share.

Agree with this approach.

> Some ideas to consider:

Perhaps we could ask @dgash to help with rethinking how best to structure the documentation? We haven't really stepped back to think how this could be improved until now so could be a good change to get it right.

**Comment by surma** [comment]
+1 on pulling in @dgash.

**Comment by sgomes** [comment]
Yup, absolutely! Should have mentioned that from the beginning :)

**Comment by mrhoric** [comment]
now you are update the website? i can not open it

**Comment by sgomes** [comment]
@mrhoric Nope, we haven't touched anything. It could be a problem on your end, as I'm not seeing any issues.

**Comment by Garbee** [comment]
Just took a stab and updating the textfield component docs to be a bit more streamlined. Available [on this google doc](https://docs.google.com/document/d/1xvUWYfJvwlI2Rvnim7KUZCLpjuh6rkR_6N5U7MrhvZI/edit?usp=sharing).

This example drops the step-by-step instructions. Instead opting to provide a quick concise paragraph on usage, followed up with examples of different combinations. This will help trim down a lot of the reading (and maintenance). Developers actively working can get in-and-out more quickly. Further, true JS API section consisting of the available methods and, if provided, any custom events.

**Comment by Garbee** [comment]
Another interesting idea... What if we went to saying, at least for certain things like textfield, that the JS is optional. By documenting all the available classes that affect the element (as we should) we can then let developers know what does what and they can handle creating their own JS logic if they wish. This way our JS is "opt-in" where now it is currently required due to the lack of full documentation for each component.
